pyproject.toml

```diff
@@ -25,8 +25,7 @@ readme = "README.md"
 [tool.poetry.dependencies]
 python = ">=3.10.0"
 pyspark = ">=3.5.0"
-numpy = ">=1.16.4"
-s3fs = "2022.11.0"
+numpy = "^1.26.4"
```

Comment:
Commit said lock poetry, meant numpy, ^ means to not go above a major version. >2 seems to break the repo. > requirements should generally probably be avoided